kornia/augmentation/random_generator/_2d/perspective.py

```diff
@@ -67,10 +69,12 @@ def forward(self, batch_shape: torch.Size, same_on_batch: bool = False) -> Dict[
         rand_val: torch.Tensor = _adapted_rsampling(start_points.shape, self.rand_val_sampler, same_on_batch).to(
             device=_device, dtype=_dtype
         )
-
-        pts_norm = torch.tensor([[[1, 1], [-1, 1], [-1, -1], [1, -1]]], device=_device, dtype=_dtype)
-
-        end_points = start_points + factor * rand_val * pts_norm
+        if not self.area_preserving:
+            pts_norm = torch.tensor([[[1, 1], [-1, 1], [-1, -1], [1, -1]]], device=_device, dtype=_dtype)
+            offset = factor * rand_val * pts_norm
+        else:
+            offset = 2 * factor * (rand_val - 0.5)
```

Comment:
it seems not guaranteed to be inside the box, right? I am thinking of some edge cases: 1) if `rand_val` is all zeros, 2) if `rand_val` is all ones.

Anyway to make it more promising? One simple way can be using a Gaussian instead of Uniform, or using pts_norm like `[[-1, -1], [1, -1], [1, 1], [-1, 1]]`.

Reply:
@Parskatt can you take a look at this? I am wondering why choosing this formula, would this giving a better training results?

Reply:
> it seems not guaranteed to be inside the box, right? I am thinking of some edge cases: 1) if rand_val is all zeros, 2) if rand_val is all ones.

Yes, it is not guaranteed to be inside the box, if it was it would always shrink the image by definition. If rand_val is all ones we have no issue, it will be a simple translation of the image. The degenerate case is if we have factor = 1 and randval so that each point goes towards the center. This will make the image into a point. However, this could also happen with the previous implementation.

> Anyway to make it more promising? One simple way can be using a Gaussian instead of Uniform, or using pts_norm like [[-1, -1], [1, -1], [1, 1], [-1, 1]]

Yes, gaussian could make sense to get it circle symmetric, but I don't think it matters much. The problem with pts_norm is that it restricts the possible orientations (atleast if using positive uniform). I will check some other repos to see their implementations.

Reply:
![image](https://user-images.githubusercontent.com/22053118/156029900-cc7ebb95-4ce0-4324-ae0d-f0cab7433ac5.png)
Taken from WarpC https://arxiv.org/abs/2104.03308 
I think a gaussian (it should probably be clipped to ensure that the homography makes sense), makes sense, but it is not really necessary.

Reply:
There are some more advanced versions, e.g.

https://github.com/rpautrat/SuperPoint/blob/313a0a115961bc2a98a5ed1d7bde33fa99542398/superpoint/models/homographies.py#L117 

which seem to generate homographies based on only rotations and translations.

Reply:
Basically, sampling "reasonable" homographies can be done in a lot of ways. The important thing, I think, is to make sure that the images basically stay the same size, and dont stretch or distort the image too severly. (typically it should be convex, which should be fine with the proposed implementation).

Reply:
@Parskatt Thanks for the contents. I am more inclined to have a function or formula from a standard library or research papers that we can cite from. It is absolutely fine even though it is totally different from what we have now. In that case, looks like the SuperPoint is one is our current best option.

Not an expert in homographies, @edgarriba @ducha-aiki @lferraz any insights in this?

Reply:
I like the idea. 
Also could be extended (may be in another PR) from "area preserving" option to "generate in this area ratio range".